source/nordic-sdk/components/softdevice/s130/headers/nrf_svc.h

```diff
@@ -49,7 +49,7 @@
   { \
     __asm( \
         "svc %0\n" \
-        "bx r14" : : "I" ((uint32_t)number) : "r0" \
```

Comment:
This is a very old issue from SDK 5.2. See this post on the devzone. The cast to `uint16_t` works for me. https://devzone.nordicsemi.com/question/6950/upgrading-to-sdk-520-breaks-build/

Btw, this is still a known issue in SDK 10, see my answer here: https://devzone.nordicsemi.com/question/57411/what-are-nrf51-sdk-100-known-issues/?answer=58295#post-id-58295